### PR TITLE
Test that CreatorProvider.Get returns valid data for all BundleTypes

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

The new test `Get_ReturnsValidData_ForAllBundleTypes` will ensure that the `Get` method of `CreatorProvider` will return valid data corresponding to all `BundleType`s.

BundleCreator object creation is a little tricky, hence the two private methods for the `Arrange` part.

## References

* #0000

### CODE REVIEW

File: docs/business-processes/validations.md [hunk 1]
```diff
@@ -4,7 +4,6 @@ The following validation rules are currently implemented in the charge domain.
 
 |**Rule number**|**Description**|**Rejection code**|**Charge Types**|**Charge Links**|
 |:-|:-|:-|:-|:-|
-|VR.009|The energy business process of a metering point is mandatory|D02|All|N/A|
```
Comment (lasrinnil) [inline docs/business-processes/validations.md:7]: Why is this removed?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Bundling/BundleCreatorProvider.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public IBundleCreator Get(DataBundleRequestDto request)
                 BundleType.ChargeLinkConfirmationDataAvailable => _bundleCreators[typeof(BundleCreator<AvailableChargeLinksReceiptData>)],
                 // RSM-031 CIM XML 'NotifyBillingMasterData' requests
                 BundleType.ChargeLinkDataAvailable => _bundleCreators[typeof(BundleCreator<AvailableChargeLinksData>)],
-                // RSM-033 CIM XML 'RejectRequestChangeBillingMasterData' rejections
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Bundling/BundleCreatorProvider.cs:53]: I take it, that the RejectRequestChangeBillingMasterData is a RSM-30?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/MessageHub/Infrastructure/Bundling/BundleSenderTests.cs [hunk 3]
```diff
@@ -76,5 +86,76 @@ public class BundleSenderTests
...
+                switch (bundleType)
+                {
+                    case BundleType.ChargeDataAvailable:
+                        bundleCreators.Add(CreateBundleCreator<AvailableChargeData>(storageHandler));
+                        break;
+                    case BundleType.ChargeConfirmationDataAvailable:
+                        bundleCreators.Add(CreateBundleCreator<AvailableChargeReceiptData>(storageHandler));
+                        break;
+                    case BundleType.ChargeRejectionDataAvailable:
+                        // BundleCreator<AvailableChargeReceiptData> already added
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/MessageHub/Infrastructure/Bundling/BundleSenderTests.cs:134]: How are these already added?

**Review by lasrinnil — APPROVED** [review]
lgtm

File: docs/business-processes/validations.md (hunk 1 above)
Comment (HenrikSommer) [reply docs/business-processes/validations.md:7]: The validation rule was removed in an earlier pull request.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Bundling/BundleCreatorProvider.cs (hunk 2 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.MessageHub/Infrastructure/Bundling/BundleCreatorProvider.cs:53]: Yes, exactly.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/MessageHub/Infrastructure/Bundling/BundleSenderTests.cs (hunk 3 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/MessageHub/Infrastructure/Bundling/BundleSenderTests.cs:134]: The same type is added for ChargesConfirmationDataAvailable.
